src-tauri/petals/build_petals_sidecar.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# Set the paths to the Python script, requirements file, and bash script
+TAURI_PATH="$(pwd)/src-tauri"
+PYTHON_SCRIPT_PATH="${TAURI_PATH}/petals/run_petals.py"
+REQUIREMENTS_FILE_PATH="${TAURI_PATH}/petals/requirements.txt"
+BUILD_SIDE_SCRIPT_PATH="${TAURI_PATH}/petals/build_petals_sidecar.sh"
+DIST_PATH="${TAURI_PATH}/bin/python"
+
+# Create a Python virtual environment
+python3 -m venv prem
+
+# Activate the virtual environment
+source prem/bin/activate
+
+# Install the necessary packages
+pip install -r $REQUIREMENTS_FILE_PATH
+
+# Package the Python script
+pyinstaller --onefile $PYTHON_SCRIPT_PATH --distpath $DIST_PATH --clean -n petals-x86_64-apple-darwin
```

Comment:
This will not work on all architectures, please build one for each architecture

```suggestion
pyinstaller --onefile $PYTHON_SCRIPT_PATH --distpath $DIST_PATH --clean -n petals-aarch64-apple-darwin

Reply:
dont trust, verify (its the AI) https://chat.openai.com/share/2a996ae8-abe3-4b50-a13a-3a6c8c41afbd

Reply:
dont trust, verify (its the AI) https://chat.openai.com/share/2a996ae8-abe3-4b50-a13a-3a6c8c41afbd

Reply:
If we manage to make an universal binary, we should them call it `petals-universal-apple-darwin` otherwsie when you will be up to compile

```
       Error failed to bundle project: Failed to copy external binaries: Failed to copy external binaries: path matching bin/python/petals-universal-apple-darwin not found.: path matching bin/python/petals-universal-apple-darwin not found.
```

Reply:
if not possible to have universal binary, we must also change how we build the desktop app and file naming (splitting Apple SIlicon and Apple Intel) on website, documentations etc..

Reply:
universal bin [may or may not be possible](https://pyinstaller.org/en/stable/feature-notes.html#macos-multi-arch-support)

```suggestion
pyinstaller --onefile $PYTHON_SCRIPT_PATH --distpath $DIST_PATH --clean --target-arch universal2 -n petals-universal-apple-darwin
```